Add dockerfile for android NDK build

MoroccanMalinois opened the pull request:
requires #2138

oleksandr-yefremov commented:
Hi @MoroccanMalinois! Thanks for posting this PR, it would be great to be able to use Docker container, because I  mostly work on macOS. I tried to follow [this instruction](https://github.com/MoroccanMalinois/monero-core/tree/android_dockerfile/android) in your fork, building base environment works fine, but building GUI fails with following error:

```
make[3]: Entering directory '/opt/android/qt5/qtbase/src/corelib'
/opt/android/android-ndk-r15/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ -c -target armv7-none-linux-androideabi -fstack-protector-strong -DANDROID -march=armv7-a -mfloat-abi=softfp -fno-builtin-memmove --sysroot=/opt/android/android-ndk-r15/platforms/android-21/arch-arm/ -mfpu=vfp4 -O2 -Os -mthumb -std=c++1z -fvisibility=hidden -fvisibility-inlines-hidden -Wall -W -Wdate-time -D_REENTRANT -fPIC -DQT_NO_USING_NAMESPACE -DQT_NO_FOREACH -DQT_BUILD_CORE_LIB -DQT_BUILDING_QT -DQT_NO_CAST_TO_ASCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT -DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -DQT_NO_DEBUG -Wall -Wextra -Werror -Woverloaded-virtual -Wshadow -Wundef -Wfloat-equal -Wnon-virtual-dtor -Wpointer-arith -Wformat-security -Wno-long-long -Wno-variadic-macros -pedantic-errors -Wchar-subscripts -Wold-style-cast -Wdouble-promotion -I. -Iglobal -I../3rdparty/harfbuzz/src -I../3rdparty/md5 -I../3rdparty/md4 -I../3rdparty/sha3 -I../3rdparty/double-conversion/include -I../3rdparty/double-conversion/include/double-conversion -I../3rdparty/forkfd -I../../include -I../../include/QtCore -I../../include/QtCore/5.8.1 -I../../include/QtCore/5.8.1/QtCore -I.moc -I../3rdparty/pcre -isystem /opt/android/android-ndk-r15/sources/cxx-stl/llvm-libc++/include -isystem /opt/android/android-ndk-r15/platforms/android-21/arch-arm/usr/include -I../../mkspecs/android-clang-libc -DQT_NO_CAST_TO_ASCII=1 -DQT_NO_CAST_FROM_ASCII=1 -DQT_STRICT_ITERATORS -DQT_NO_URL_CAST_FROM_STRING=1 -DQT_NO_CAST_FROM_BYTEARRAY=1 -DQT_NO_KEYWORDS=1 -DQT_USE_QSTRINGBUILDER -DQT_USE_FAST_OPERATOR_PLUS -Dsignals=int -Dslots=int -Demit=public: -Dforeach=public: -Dforever=public: -xc++ animation/qabstractanimation.h -o .obj/header_qabstractanimation.o
g++ -c -pipe -O2 -std=c++11 -fno-exceptions -Wall -W -D_REENTRANT -fPIC -DQT_UIC -DQT_NO_CAST_FROM_ASCII -DQT_NO_FOREACH -DQT_UIC_CPP_GENERATOR -DQT_USE_QSTRINGBUILDER -DQT_NO_EXCEPTIONS -DQT_NO_DEBUG -DQT_BOOTSTRAP_LIB -DQT_VERSION_STR='"5.8.1"' -DQT_VERSION_MAJOR=5 -DQT_VERSION_MINOR=8 -DQT_VERSION_PATCH=1 -DQT_BOOTSTRAPPED -DQT_NO_CAST_TO_ASCII -I. -I. -Icpp -I. -I../../../include -I../../../include/QtCore -I../../../include/QtCore/5.8.1 -I../../../include/QtCore/5.8.1/QtCore -I../../../include/QtXml -I../../../include/QtXml/5.8.1 -I../../../include/QtXml/5.8.1/QtXml -I../../../mkspecs/linux-g++ -o .obj/uic.o uic.cpp
In file included from animation/qabstractanimation.h:43:
In file included from ../../include/QtCore/qobject.h:1:
In file included from ../../include/QtCore/../../src/corelib/kernel/qobject.h:47:
In file included from ../../include/QtCore/qstring.h:1:
../../include/QtCore/../../src/corelib/tools/qstring.h:61:35: error: declaration shadows a typedef in namespace 'std::__ndk1' [-Werror,-Wshadow]
    typedef basic_string<wchar_t> wstring;
                                  ^
/opt/android/android-ndk-r15/sources/cxx-stl/llvm-libc++/include/iosfwd:195:74: note: previous declaration is here
typedef basic_string<wchar_t, char_traits<wchar_t>, allocator<wchar_t> > wstring;
                                                                         ^
1 error generated.
Makefile:1904: recipe for target '.obj/header_qabstractanimation.o' failed
make[3]: Leaving directory '/opt/android/qt5/qtbase/src/corelib'
make[3]: *** [.obj/header_qabstractanimation.o] Error 1
make[2]: *** [sub-corelib-make_first] Error 2
make[2]: *** Waiting for unfinished jobs....
Makefile:194: recipe for target 'sub-corelib-make_first' failed
g++ -Wl,--gc-sections -Wl,-O1 -o ../../../bin/uic .obj/customwidgetsinfo.o .obj/databaseinfo.o .obj/driver.o .obj/treewalker.o .obj/ui4.o .obj/validator.o .obj/cppextractimages.o .obj/cppwritedeclaration.o .obj/cppwriteicondata.o .obj/cppwriteicondeclaration.o .obj/cppwriteiconinitialization.o .obj/cppwriteincludes.o .obj/cppwriteinitialization.o .obj/main.o .obj/uic.o   -L/opt/android/qt5/qtbase/lib -lQt5Bootstrap -lpthread
make[3]: Leaving directory '/opt/android/qt5/qtbase/src/tools/uic'
make[2]: Leaving directory '/opt/android/qt5/qtbase/src'
make[1]: *** [sub-src-make_first] Error 2
Makefile:46: recipe for target 'sub-src-make_first' failed
make[1]: Leaving directory '/opt/android/qt5/qtbase'
make: *** [module-qtbase-make_first] Error 2
Makefile:73: recipe for target 'module-qtbase-make_first' failed
The command '/bin/sh -c cd qt5/qttools     && git apply androiddeployqt.patch     && cd ..     && PATH=${CLEAN_PATH} make -j4     && PATH=${CLEAN_PATH} make install' returned a non-zero code: 2
```

Any ideas why this might happen would be appreciated.

oleksandr-yefremov commented:
Ok, I got it working on my side. Perhaps it will help someone else.
> ENV ANDROID_NDK_REVISION 13b

I tried several revisions of NDK, the first one to compile was 13b. Also, not sure why all this hassle about "Deployment", maybe I am missing smth, but I was able to simply do `adb install ./QtApp-debug.apk` to install the app.

MoroccanMalinois commented:
Hi @oleksandr-yefremov , yes my fault. I updated yesterday android NDK version from 14b to 15. Looks like it was not a great idea :)
Switched back to 14.

MoroccanMalinois commented:
> maybe I am missing smth

Nah. Total android noob here, didn't knew about `adb install`
